### DISCUSSION TITLE
[BUG] createDraftListing with type download returns error Internal Server Error 500 if you send shipping_profile_id as null

### DISCUSSION THREAD

**Discussion by NickMihaiu91** [etsy/open-api#688] [Q&A] [answered]
createDraftListing with type download returns error **"Internal Server Error" 500** if you send shipping_profile_id as null

In the documentation: https://developers.etsy.com/documentation/reference#operation/createDraftListing it says **shipping_profile_id** is of type: **integer >= 1 || Nullable**

If you say in the documentation that Nullable is allowed I'm expecting the code to work if I send a request with shipping_profile_id set to Null. Or at least it should give me a nice error message but the error is "Internal Server Error" 500 which doesn't say nothing about the actual issue and I suspect it's actually crashing your code which would be kind of bad.

Attached image with my listing data and response from createDraftListing
<img width="390" alt="listing-data" src="https://user-images.githubusercontent.com/5738598/206188639-9bbeb0e3-4d62-4cf5-ae49-0180c05ea71d.png">
<img width="659" alt="response-error" src="https://user-images.githubusercontent.com/5738598/206188647-c0e22d88-ace7-4a03-a863-ea46e56cad11.png">

This should be handled on the APIs side.

**Comment by etsyachristensen** [comment, marked as answer]
@NickMihaiu91 We aren't able to duplicate this error. Can you send the full payload as text rather than an image? I need to try to duplicate your experience. That said, in the OpenAPI, if a field is marked as `nullable` that means you can leave it out. Passing it in with a null value *should* work but is not recommended.

**Reply by NickMihaiu91** [reply]
Hi @etsyachristensen 

Thank you for looking into it. For me it's reproducible every time and I have tried with several different info in the payload. 
This is one example JSON payload that breaks the API for me:
`{"title":"Personalized First Year As A Teacher Png, 1st Year Teacher Digital Design, First Time Teacher Digital Print, Custom Name Digital Download","tags":"1st year teacher PNG,1st time teacher PNG,custom name PNG,custom name gift,1st year teacher,first time teacher,1st time teacher,custom name,digital design,digital print,digital download,personalized,first","quantity":999,"description":"Cute DIGITAL design to download and use for creating unique appreciation gifts for your dear one.\n\n\nFeatures:\n-------------\n😊 It has a TRANSPARENT background.\n\n😊 Please mind that this is a DIGITAL product; NO physical product will be shipped.\n\n😊 Designs without personalization will be ready for download instantly.\n     Designs with personalization will be ready for download in 24 to 72 hours.\n\n😊 If you would like a particular customization, please contact me.\n\n😊 If you would like it shipped on a physical product (like Greeting Card, Ceramic Ornament a.s.o.), please contact me, I can make a specific product available for you to order.\n\n\nCopyright:\n--------------\n😊 For PERSONAL USE only, no commercial use.\n\n😊 All copyright rights belong to SimilinGifts store owner.","type":"download","materials":[],"shipping_profile_id":null,"shop_section_id":39980370,"who_made":"i_did","when_made":"made_to_order","should_auto_renew":false,"taxonomy_id":2078,"price":6.98,"image_ids":[4416256590],"is_personalizable":true,"personalization_is_required":true,"personalization_char_count_max":256,"personalization_instructions":"Please enter the Title, the Name, the School and the Year, separated by comma:\ne.g. Mrs. Smith, Heart Elementary, 2022"}`

Just removing the `shipping_profile_id` that is currently set to Null in the above payload fixes it and I get a succes response and the listing is created successfully.

**Reply by etsyachristensen** [reply]
Ok thanks for the update. We will look into the error when it's passed in as null. It should be ignored, but clearly it's causing a problem. This may take some time to get around to though.

**Comment by baahar** [comment]
Hi Nick,

Have you tried sending your request without the shipping_profile_id attribute altogether?

**Reply by NickMihaiu91** [reply]
@baahar yes I did as I said from the beginning. And it works if you do not send `shipping_profile_id` but that's not the point. The point is that if you say in the documentation that you can send it as NULL then it should not break the entire request and return a 500 Internal Server Error.

**Reply by baahar** [reply]
Ah ok, I see your comment below the payload now. 

The error message I was getting before omitting the attribute was: 
400 - Bad Request: Expected int value for shipping_profile_id (got string)
